### PR TITLE
SSO LDAP: Add LDAP settings drawer

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**
This feature adds a Drawer to add settings for LDAP UI. This new feature includes extra configuration settings.

**Why do we need this feature?**

This is part of the implementation for LDAP UI described here - [link](https://www.figma.com/design/VxdG9sTKLBW8rtxwh05PNr/%F0%9F%94%B5-Authentication-configuration?node-id=2160-8184&t=piBBbJ3XNkLQLno8-0)

**Who is this feature for?**

IAM

**Which issue(s) does this PR fix?**:

This PR covers part of the roadmap described here - [link](https://github.com/grafana/identity-access-team/issues/714#issuecomment-2220008265)

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by mgyongyosi — APPROVED** [review]

File: public/app/features/admin/ldap/LdapDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+      </CollapsableSection>
+      <CollapsableSection label={t('ldap-drawer.attributes-section.label', 'Attributes')} isOpen={true}>
+        <Text color="secondary">
+          <Trans i18nKey="ldap-drawer.attributes-section.description">
+            Specify the LDAP attributes that map to the user&lsquo;s given name, surname, and email address, ensuring
+            the application correctly retrieves and displays user information.
+          </Trans>
+        </Text>
+        <Field htmlFor="name" label={t('ldap-drawer.attributes-section.name.label', 'Name')}>
+          <Input id="name" {...register('settings.config.servers.0.attributes.name')} />
```
Comment (ashharrison90) [inline public/app/features/admin/ldap/LdapDrawer.tsx:103]: react has a `useId()` hook that can help you generate ids: https://react.dev/reference/react/useId

problem here is that id is supposed to be unique on a page. but when the ids are so generic, like `name`, it's only a matter of time before there's more than one on a page and this linking might break

File: public/app/features/admin/ldap/LdapDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+      {
+        <TextLink style={{ fontSize: 'inherit' }} href="https://go.dev/src/crypto/tls/cipher_suites.go" external>
+          https://go.dev/src/crypto/tls/cipher_suites.go
+        </TextLink>
+      }
+    </Trans>
+  );
+
+  return (
+    <Drawer title={t('ldap-drawer.title', 'Advanced Settings')} onClose={onClose}>
```
Comment (ashharrison90) [inline public/app/features/admin/ldap/LdapDrawer.tsx:55]: think we use sentence case everywhere, so this should be `Advanced settings`. haven't checked all the other strings, there might be others

File: public/app/features/admin/ldap/LdapDrawer.tsx
Comment (ashharrison90) [inline public/app/features/admin/ldap/LdapDrawer.tsx:1]: this is quite a beefy form. you could split each section into it's own component (e.g. `LdapMiscSection`, `LdapAttributesSection`, etc.) and pass the `useFormContext` methods through to the child components. i don't think you _have_ to, but it's right on the edge 😅 if it got any bigger i would definitely recommend that.

**Review by ashharrison90 — COMMENTED** [review]
mostly just nits. the only thing i think you definitely should do is remove the port number/timeouts from the translations 👍

**Pull request merged**